hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -147,14 +147,13 @@ public void run() {
           }
         }
       } catch (IOException e) { // stream related
-        if (sleepMultiplier < maxRetriesMultiplier) {
-          LOG.debug("Failed to read stream of replication entries: " + e);
-          sleepMultiplier++;
-        } else {
-          LOG.error("Failed to read stream of replication entries", e);
-          handleEofException(e);
+        if (!handleEofException(e)) {
```

Comment:
let's reset the sleep multiplier it EOF was handled.
```java
if (handleEofException(e))
{
    sleepMultiplier = 1;
}
```

Reply:
Thank @sandeepvinayak for reviewing.
How about keeping resetting `sleepMultiplier` only in [there](https://github.com/apache/hbase/blob/1a703f48e90b985b00aeee1f1de6c876e60acd68/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java#L143) where read WAL successfully? What I'm thinking is that `handleEofException` just deals with exceptions and we still need to retry to get back to the correct logic. We reset `sleepMultiplier` only after the logic getting back correct?

Reply:
yeah, but in handleEofException, we deque the bad(or empty) log file and do the repair work. So if we did the repair work why not reset the sleep multiplier?

Reply:
OK, I modified the code as your comments.